BasicSample/Assets/Interaction/Scripts/ApplyHaptics.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+    /// <summary>
+    /// Checks for haptics support in the controller and if supported, it sends an impulse when the trigger is pressed.
+    /// </summary>
+    public class ApplyHaptics : MonoBehaviour
+    {
+        private readonly List<InputDevice> controllerHapticDevices = new List<InputDevice>();
+
+        private void InitializeDevices()
+        {
+            controllerHapticDevices.Clear();
+            List<InputDevice> controllerInputDevices = new List<InputDevice>();
+
+            // Get all controller input devices
+            InputDevices.GetDevicesWithCharacteristics(InputDeviceCharacteristics.Controller, controllerInputDevices);
+
+            // Filter devices with haptics capabilities
+            foreach (InputDevice controller in controllerInputDevices)
+            {
+                if (controller.TryGetHapticCapabilities(out HapticCapabilities hapticCapabilities) && hapticCapabilities.supportsImpulse)
+                {
+                    controllerHapticDevices.Add(controller);
+                }
+            }
+        }
+
+        private void OnDeviceConnected(InputDevice device)
+        {
+            InitializeDevices();
+        }
+
+        private void OnDeviceDisconnected(InputDevice device)
+        {
+            InitializeDevices();
+        }
+
+        private void Start()
+        {
+            InitializeDevices();
```

Comment:
I haven't loaded up the scene yet, but how does the toggle button for this feature work? Does it enable/disable this script directly? If so, it may be worth moving `InitializeDevices()` into `OnEnable()`, to guarantee any device state changes that were missed while this script wasn't active are detected

Reply:
The toggle calls SetActive on the game object

Reply:
Okay cool, then yeah `OnEnable` seems reasonable so it gets called each time

Reply:
Done